File: dmoj/ccc00s4/ccc00s4.py
import sys
input = sys.stdin.readline

d = int(input())
n = int(input())

clubs = [int(input()) for i in range(n)]

INF = 1 << 30
dp = [INF]*(d+1)

dp[0] = 0
for pos in range(1, d+1):
    for a in clubs:
        if a <= pos:
            dp[pos] = min(dp[pos], dp[pos-a] + 1)

if dp[d] == INF:
    print("Roberta acknowledges defeat.")
else:
    print("Roberta wins in " + str(dp[d]) + " strokes.")

File: dmoj/ccc00s4/ccc00s4.cpp
#include <iostream>
#include <algorithm>
using namespace std;

const int INF = 1 << 30;
const int MD = 5300;
const int MN = 35;
int d, n;
int dp[MD];
int clubs[MN];

int main(){
    cin.sync_with_stdio(0); cin.tie(0);
    cin >> d >> n;
    for(int i = 0; i < n; i++) cin >> clubs[i];
    for(int pos = 1; pos <= d; pos++){
        dp[pos] = INF;
        for(int i = 0; i < n; i++){
            if(clubs[i] <= pos){
                dp[pos] = min(dp[pos], dp[pos-clubs[i]] + 1);
            }
        }
    }

    if(dp[d] == INF) printf("Roberta acknowledges defeat.\n");
    else printf("Roberta wins in %d strokes.\n", dp[d]);
}
